library/src/test/java/io/voucherify/android/client/module/ValidationTest.java

```diff
@@ -0,0 +1,124 @@
+package io.voucherify.android.client.module;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.reactivex.schedulers.Schedulers;
+import io.voucherify.android.client.api.VoucherifyApi;
+import io.voucherify.android.client.model.OrderItem;
+import io.voucherify.android.client.model.VoucherResponse;
+import okhttp3.Request;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ValidationTest {
```

Comment:
I have a problem to understand below tests. We mock the API and create the validation object. So far, so good but then in these methods we invoke validateVoucher twice, one on validation object, then on api which is mocked and we check if this method was invoked or not where we are invoking it directly so the test will pass for sure.